app/controllers/members_controller.rb

```diff
@@ -0,0 +1,52 @@
+class MembersController < ApplicationController
+  before_action :authenticate_user!
```

Comment:
Let's only require authentication for `make_admin`, `remove_admin`, and adding/removing users. Anyone (even people who are logged out) should be able to view a group.